File: LongestConsecutiveSequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> s = Arrays.stream(nums).boxed().collect(Collectors.toSet());
        int longest = 0;
        int len = 0;

        for (int num: nums){
            if(!s.contains(num - 1)){
                len++;

                while(s.contains(num+len)){
                    len++;
                }
                longest = Math.max(longest, len);
                len = 0;
            }
        }
        return longest;
    }
}
// Time: O(n)
// Space: O(n)

File: LongestConsecutiveSequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set(nums)
        longest = 0
        ans = 0

        for num in nums:
            if (num - 1) not in s:
                ans += 1
                
                while num + ans in s:
                    ans += 1
                
                longest = max(longest, ans)
                ans = 0
        
        return longest
        
    # Time: O(n)
    # Space: O(n)
